compiler/src/dmd/errors.d

```diff
@@ -476,7 +648,7 @@ private extern(C++) void verrorReport(const SourceLoc loc, const(char)* format,
             info.headerColor = Classification.error;
             if (global.params.v.messageStyle == MessageStyle.sarif)
             {
-                generateSarifReport(loc, format, ap, info.kind, false);
+                diagnostics ~= Diagnostic(loc, formattedMessage, kind);
```

Comment:
just modify `generateSarifReport` report to do this, and make the formatting changes there